frontend/summary/dataPivot/store.js

```diff
@@ -0,0 +1,84 @@
+import _ from "lodash";
+import {action, observable, toJS} from "mobx";
+
+import {
+    moveArrayElementUp,
+    moveArrayElementDown,
+    deleteArrayElement,
+} from "shared/components/EditableRowData";
+import {NULL_CASE, OrderChoices} from "./shared";
+
+class SortStore {
+    constructor(rootStore) {
+        this.rootStore = rootStore;
+        this.settings = this.rootStore.dp.settings.sorts;
+    }
+
+    @observable settings = null;
+
+    @action.bound createNew() {
+        this.settings.push({
+            field_name: NULL_CASE,
+            order: OrderChoices.asc,
+            custom: null,
+        });
+        this.rootStore.sync();
+    }
+    @action.bound moveUp(idx) {
+        moveArrayElementUp(this.settings, idx);
+        this.rootStore.sync();
+    }
+    @action.bound moveDown(idx) {
+        moveArrayElementDown(this.settings, idx);
+        this.rootStore.sync();
+    }
+    @action.bound delete(idx) {
+        deleteArrayElement(this.settings, idx);
+        this.rootStore.sync();
+    }
+    @action.bound updateElement(idx, field, value) {
+        this.settings[idx][field] = value;
+        this.rootStore.sync();
+    }
+    @action.bound updateOrder(idx, value) {
+        if (value === OrderChoices.custom) {
+            this.settings[idx].custom = this.getSortItems(idx);
+        } else {
+            this.settings[idx].custom = null;
+        }
+        this.settings[idx].order = value;
+        this.rootStore.sync();
+    }
+    @action.bound changeOrder(idx, oldIndex, newIndex) {
+        const items = _.cloneDeep(toJS(this.settings[idx].custom)),
+            item = items.splice(oldIndex, 1)[0];
+        items.splice(newIndex, 0, item);
+        this.settings[idx].custom = items;
+        this.rootStore.sync();
+    }
+    getSortItems(idx) {
+        const key = this.settings[idx].field_name;
+        if (key === NULL_CASE) {
+            return [];
+        }
+        return _.chain(this.rootStore.dp.data)
+            .map(key)
+            .uniq()
+            .sort()
+            .value();
+    }
+}
+
+class Store {
+    constructor(dp) {
+        this.dp = dp;
+        this.sortStore = new SortStore(this);
+    }
+    overrideRefresh = null;
```

Comment:
Remove?

Reply:
Good catch. Let me add back the override refresh mechanism to  he store. I had it working and stripped it out.

Reply:
Added this back; it now behaves the same as before; good catch!